src/h5web/vis-packs/core/matrix/GridSettingsContext.tsx

```diff
@@ -1,15 +1,12 @@
-import type Complex from 'complex.js';
 import { ReactNode, createContext } from 'react';
-import type { Size } from '../models';
+import type { PrimitiveType } from '../../../providers/models';
+import type { PrintableType, Size } from '../models';
 
 interface GridSettings {
   cellSize: Size;
   rowCount: number;
   columnCount: number;
-  valueAccessor: (
-    row: number,
-    col: number
-  ) => number | string | boolean | Complex;
+  valueAccessor: (row: number, col: number) => PrimitiveType<PrintableType>;
```

Comment:
While using PrimitiveType<PrintableType> is handy, this is not awfully readable. Glad to have your input on this.

Reply:
Good idea on using `PrintableType` 👍

It's true that the `Type` suffix is more for our adaptation of HDF5 types. Perhaps just rename it to `Primitive`?